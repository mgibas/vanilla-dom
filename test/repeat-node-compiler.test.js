const compiler = require('../src/repeat-node-compiler')
let tagNodeCompiler = require('../src/tag-node-compiler')
let repeatNodeUpdateCompiler = require('../src/repeat-node-update-compiler')
let childrenCompiler = jest.fn()
tagNodeCompiler.compile = jest.fn()
repeatNodeUpdateCompiler.compile = jest.fn()
childrenCompiler.mockImplementation((a,b) => `CHILDREN_COMPILER_PART('${a}', '${b}')`)
tagNodeCompiler.compile.mockImplementation((a,b,c) => `TAG_NODE_COMPILER_PART('${a}', '${b}', '${c===childrenCompiler}')`)
repeatNodeUpdateCompiler.compile.mockImplementation((a,b,c,d) => `REPEAT_NODE_UPDATE_COMPILER_PART('${a}', '${b}', '${c===childrenCompiler}', '${d}')`)

describe('repeat-node-compiler', () => {
  let options = {state:'st'}

  describe('on compile', () => {
    it('simple token', () => {
      expect(compiler.compile('parent', {
        attribs: { 'repeat-for': '{{st.items}}' }
      }, childrenCompiler, options))
        .toMatchSnapshot();
    })
    it('custom as and index', () => {
      expect(compiler.compile('parent', {
        attribs: { 'repeat-for': '{{st.items}}', 'repeat-as': 'rows', 'repeat-index': 'z'}
      }, childrenCompiler, options))
        .toMatchSnapshot();
    })
  })
})
