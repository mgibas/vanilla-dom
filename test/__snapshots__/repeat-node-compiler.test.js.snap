// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeat-node-compiler on compile custom as and index 1`] = `
"(() => {
  let _nodes = [];
  let rows = (() => {
    try {
      return st.items;
    } catch (err) {
      return;
    }
  })() || [];
  let _createChildren = z => {
    node = TAG_NODE_COMPILER_PART('parent', '[object Object]', 'true');
    _nodes.push(node);
  };
  REPEAT_NODE_UPDATE_COMPILER_PART('parent', '[object Object]', 'false', '[object Object]');
  for (let z = 0; z < rows.length; z++) {
    _createChildren(z);
  }
})();"
`;

exports[`repeat-node-compiler on compile simple token 1`] = `
"(() => {
  let _nodes = [];
  let items = (() => {
    try {
      return st.items;
    } catch (err) {
      return;
    }
  })() || [];
  let _createChildren = i => {
    node = TAG_NODE_COMPILER_PART('parent', '[object Object]', 'true');
    _nodes.push(node);
  };
  REPEAT_NODE_UPDATE_COMPILER_PART('parent', '[object Object]', 'false', '[object Object]');
  for (let i = 0; i < items.length; i++) {
    _createChildren(i);
  }
})();"
`;
